index.js

```diff
@@ -57,8 +76,21 @@ function Person() {
         + The `drive` method should return a string "I ran out of fuel at x miles!" x being `odometer`.
 */
 
-function Car() {
-
+function Car(model, milesPerGallon) {
+  this.model = model;
+  this.milesPerGallon = milesPerGallon;
+  this.tank = 0;
+  this.odometer = 0;
+}
+Car.prototype.fill = function (gallons){
+  this.tank = this.tank + gallons;
+}
+Car.prototype.drive = function(distance) {
+  this.odometer = this.odometer + distance;
+  this.tank = this.tank--;
+  if (this.tank = 0){
+    return `I ran out of fuel at ${this.odometer} miles!`
+  }
 }
```

Comment:
Good attempt at the stretch, it's missing a few pieces (read the directions more carefully on the sprint!).  But this is enough to get you a 3.